proto/build.rs

```diff
@@ -11,131 +11,151 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::env;
-use std::fs::{self, File};
-use std::io::Write;
-use std::path::{Path, PathBuf};
-
-use grpcio_compiler::codegen as grpc_gen;
-use prost::Message;
-use prost_types::FileDescriptorSet as ProstFileDescriptorSet;
-use protobuf::compiler_plugin::GenResult;
-use protobuf::descriptor::{FileDescriptorProto, FileDescriptorSet};
-use protobuf_codegen::{self as pb_gen, Customize};
-
-fn write_files<W: Write>(
-    results: impl Iterator<Item = (String, Vec<u8>)>,
-    output_dir: &Path,
-    mut out: W,
-) {
-    if !output_dir.exists() {
-        fs::create_dir_all(output_dir).unwrap();
+fn main() {
+    for package in &["testing", "example", "health"] {
+        protobuf::compile(format!("{}.desc", package), package);
```

Comment:
Why not use one line instead? So only one of prost-codec and protobuf-codec will be activated.

Reply:
Hmm, maybe could, we need to support generating for both codecs, but I think because of the complex cfgs, we'll only use one anyway...

Reply:
Use only one line can only generate what's necessary. And following the modules can also be simplified.